### ISSUE TITLE
Support gradient accumulation in ExpertBackend

### ISSUE THREAD

**Issue by mryab** [learning-at-home/hivemind#198]
Larger Transformer models are trained with larger batches, it's probably beneficial to accumulate gradients from several backward requests before making a step. It can be implemented in `ExpertBackend.apply_gradients()`, and the number of processed examples is already available.

The tricky part is to implement loss averaging correctly: since we might have batches of different sizes (at least at the server side), we might need to scale the gradients from each batch according to its relative size.
